numpy-stubs/__init__.pyi

```diff
@@ -393,14 +394,39 @@ class number(generic):
 
 class bool_(_real_generic): ...
 class object_(generic): ...
-class datetime64(_real_generic): ...
+
+class datetime64:
```

Comment:
Motivation behind removing inheritance of `_real_generic`: inheritance allowed code like this to pass `mypy`
```
np.timedelta64(0) - np.datetime64(0, 'D')
```